### PR TITLE
1982: ama only push job testing

### PR DESCRIPTION
**Pull request by barshirtcliff**
This is unit tests for half of the new functionality added in https://github.com/department-of-veterans-affairs/caseflow/pull/16652

### Description
unit tests for:
- [X] ama-only-mode does not prevent distribution of tied legacy cases in the push job
- [x] ama-only-mode does not prevent distribution of ama cases in the push job
- [x] ama-only-mode does prevent distribution of genpop legacy cases in the push job

### Acceptance Criteria
- [x] Code compiles correctly
- [x] Unit tests pass

### CODE REVIEW

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb [hunk 1]
```diff
@@ -117,26 +119,63 @@ def to_judge_hash(arr)
...
+          bfdloout: 3.days.ago,
+          bfbox: nil,
+          folder: build(:folder, tinum: "1801005", titrnum: "923456790S")
+        )
+        create(
+          :case_hearing,
+          :disposition_held,
+          folder_nr: vacols_case.bfkey,
+          hearing_date: 5.days.ago.to_date,
+          board_member: judge.vacols_attorney_id
```
Comment (leikkisa) [inline spec/jobs/push_priority_appeals_to_judges_job_spec.rb:141]: Wow, you guys had to dig into VACOLS to figure out how judges are tied to cases.

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb [hunk 2]
```diff
@@ -117,26 +119,63 @@ def to_judge_hash(arr)
...
+        appeal = create(
+          :appeal,
+          :ready_for_distribution,
+          :advanced_on_docket_due_to_age,
+          uuid: "bece6907-3b6f-4c49-a580-6d5f2e1ca65d",
+          docket_type: Constants.AMA_DOCKETS.hearing
+        )
+        most_recent = create(:hearing_day, scheduled_for: 1.day.ago)
+        hearing = create(:hearing, judge: nil, disposition: "held", appeal: appeal, hearing_day: most_recent)
+        hearing.update!(judge: judge)
```
Comment (leikkisa) [inline spec/jobs/push_priority_appeals_to_judges_job_spec.rb:153]: Does putting this in the above line cause issues?

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb [hunk 3]
```diff
@@ -243,6 +283,13 @@ def to_judge_hash(arr)
         expect(distributed_cases.count).to eq target_distributions
       end
     end
+
+    it "only distributes ama cases to ama-only judge" do
```
Comment (leikkisa) [inline spec/jobs/push_priority_appeals_to_judges_job_spec.rb:287]: The wording is tricky! I could see this meaning "AMA cases only get distributed to ama-only judges" which isn't quite true - though AMA-only judges will only get distributed the AMA cases.

**Review by leikkisa — APPROVED** [review]
Only a couple of minor comments, looks like things are working as expected. I see what you're saying about how the tests are structured, yuck.

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb (hunk 1 above)
Comment (barshirtcliff) [reply spec/jobs/push_priority_appeals_to_judges_job_spec.rb:141]: yep... pretty painful.

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb (hunk 3 above)
Comment (barshirtcliff) [reply spec/jobs/push_priority_appeals_to_judges_job_spec.rb:287]: yeah...  agreed.  I may circle back and update this wording.

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb (hunk 2 above)
Comment (barshirtcliff) [reply spec/jobs/push_priority_appeals_to_judges_job_spec.rb:153]: I'm not sure what you mean.

**Pull request merged**